pwndbg/commands/misc.py

```diff
@@ -44,7 +45,8 @@ def errno_(err) -> None:
                 if errno_loc_gotplt is None or pwndbg.gdblib.vmmap.find(
                     pwndbg.gdblib.memory.pvoid(errno_loc_gotplt)
                 ):
-                    err = int(gdb.parse_and_eval("*((int *(*) (void)) __errno_location) ()"))
+                    with lock_scheduler():
+                        err = int(gdb.parse_and_eval("*((int *(*) (void)) __errno_location) ()"))
```

Comment:
Do I understand this correctly that if you call a function like this on behalf of the target in GDB, then if you have other threads those other threads are also resumed for the time this function is executed?

Wow, this is really surprising to the degree I could say that maybe this is even a GDB bug, but on the other hand maybe it makes sense to allow for such behavior, especially that the called function itself may need to interact with other threads, huh.

I think we need to document this behavior/pattern better to make sure any future readers of this code understands exactly why we do need that.

Alternatively, we can also extract the `gdb.parse_and_eval` which will perform a call to a `pwndbg.gdblib.*` function which will use the `lock_scheduler` and we could name and document that function in a descriptive way. We could also put that info into DEVELOPERS.md, as this is really important.

Reply:
What @gsingh93 proposed, where we call `set scheduler-locking on` in `ArgparsedCommand` before executing a command also seems to make sense. Though, it would be great if we can optimize it and only call it if needed.

(We eventually want the commands to be working faster if possible, since they may be used for scripting GDB)

Reply:
> Do I understand this correctly that if you call a function like this on behalf of the target in GDB, then if you have other threads those other threads are also resumed for the time this function is executed?

Yes, you are correct.

> Alternatively, we can move the gdb.parse_and_eval to a pwndbg.gdblib.* function

This is a good idea.
What function name would you prefer for this function, and which `pwndbg.gdblib.*` module should I add it to?
cc: @gsingh93 
(I'm not good at naming functions and organizing them into different modules xD)

>  We could also put that info into DEVELOPERS.md, as this is really important.

Sure, I'll add that info into `DEVELOPERS.md`.

Reply:
> What function name would you prefer for this function, and which pwndbg.gdblib.* module should I add it to?

I don't think we have an existing module that's a good place for this. And I'm honestly not sure what the module should be called, sorry 😛 

Maybe `parse_and_eval_with_scheduler_lock` for the name? The verbosity is intentional, it should be clear and stick out when we need to use it. But open to other suggestions.

Reply:
> Maybe `parse_and_eval_with_scheduler_lock` for the function name? The verbosity is intentional, it should be clear and stick out when we need to use it. But open to other suggestions.

Ok, thanks!
I put it into `pwndbg.gdblib.sheduler` for now.
Let me know if there’s another good place or good name for the module :p